compiler/rustc_mir/src/borrow_check/diagnostics/mutability_errors.rs

```diff
@@ -241,6 +244,39 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                     format!("mut {}", self.local_names[local].unwrap()),
                     Applicability::MachineApplicable,
                 );
+                let tcx = self.infcx.tcx;
+                // point to span of upvar making closure call require mutable borrow
+                if let ty::Closure(id, _) = the_place_err.ty(self.body, tcx).ty.kind() {
```

Comment:
This looks almost identical to the code you added later on this in this file. Could you factor out the common parts into a new closure/method?

Reply:
@Aaron1011 I've factored the code out.